client/src/cmdhf14b.c

```diff
@@ -2573,44 +2694,39 @@ int infoHF14B(bool verbose, bool do_aid_search) {
 // get and print general info about all known 14b chips
 int readHF14B(bool loop, bool verbose) {
     bool found = false;
+    int res = PM3_SUCCESS;
     do {
         found = false;
 
         // try std 14b (atqb)
         found |= HF14B_std_reader(verbose);
-        if (found && loop)
-            continue;
-        else if (found)
-            return PM3_SUCCESS;
+        if (found)
+            goto plot;
 
         // try ST Microelectronics 14b
         found |= HF14B_st_reader(verbose);
-        if (found && loop)
-            continue;
-        else if (found)
-            return PM3_SUCCESS;
+        if (found)
+            goto plot;
 
         // Picopass
-        found |= HF14B_picopass_reader(verbose) ;
-        if (found && loop)
-            continue;
-        else if (found)
-            return PM3_SUCCESS;
+        found |= HF14B_picopass_reader(verbose);
+        if (found)
+            goto plot;
 
         // try ASK CT 14b
         found |= HF14B_ask_ct_reader(verbose);
-        if (found && loop)
-            continue;
-        else if (found)
-            return PM3_SUCCESS;
+        if (found)
+            goto plot;
 
         // try unknown 14b read commands (to be identified later)
         // could be read of calypso, CEPAS, moneo, or pico pass.
         found |= HF14B_other_reader(verbose);
-        if (found && loop)
-            continue;
-        else if (found)
-            return PM3_SUCCESS;
+        if (found)
+            goto plot;
+plot:
+        res = handle_hf_plot();
+        if (res != PM3_SUCCESS)
```

Comment:
this will break the continues loop if  the hf_plot fails

Reply:
I think this is the same behaviour that the 14a reader plot has

Reply:
then it has the same problem :)

Reply:
what should it's behaviour be then, continue as if nothing happened or stop trying to plot?